src/main/java/software/amazon/awssdk/crt/CRT.java

```diff
@@ -152,40 +153,83 @@ public static String getCRuntime(String osIdentifier) {
             return NON_LINUX_RUNTIME_TAG;
         }
 
-        Runtime rt = Runtime.getRuntime();
-        String[] commands = {"ldd", "--version"};
-        try {
-            java.lang.Process proc = rt.exec(commands);
-
-            // the "normal" input stream of the proc is the stdout of the invoked command
-            BufferedReader stdOutput = new BufferedReader(new
-                    InputStreamReader(proc.getInputStream()));
-
-            // sometimes, ldd's output goes to stderr, so capture that too
-            BufferedReader stdError = new BufferedReader(new
-                    InputStreamReader(proc.getErrorStream()));
-
-            String line;
-            StringBuilder outputBuilder = new StringBuilder();
-            while ((line = stdOutput.readLine()) != null) {
-                outputBuilder.append(line);
+        // The system might have both musl and glibc on it:
+        // https://github.com/awslabs/aws-crt-java/issues/659
+
+        // First, check which one java is using.
+        // Run: ldd $JAVA_HOME/bin/java
```

Comment:
Yeah, that's a solid idea. Now user's can do `-Daws.crt.libc=musl` or `-Daws.crt.libc=glibc` if our autodetection fails.

I ended up documenting all our System properties in the README, since none of them were documented, and I needed to document the new one somewhere

Reply:
And yeah, there are lots of other ways to detect Musl out there, and every PR and stackoverflow  has people commenting "this still doesn't work on my machine". We can revisit other ways in the future. Just run them all one after another. But this seems good enough for now...